core/src/main/java/org/mule/processor/InvokerMessageProcessor.java

```diff
@@ -292,7 +292,7 @@ public void setMethodName(String methodName)
 
     public void setArgumentExpressionsString(String arguments)
     {
-        this.arguments = Arrays.asList(arguments.split("\\s*,\\s*"));
+        this.arguments = Arrays.asList(arguments.split("\\s*,(?=[^\\]]*(?:\\[|$))\\s*"));
```

Comment:
can you add unit tests for this? the integration test added in this PR is fine, but something more unit would be great.